Framework/DataHandling/inc/MantidDataHandling/ISISJournal.h

```diff
@@ -0,0 +1,65 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidKernel/System.h"
+
+#include <Poco/AutoPtr.h>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Poco::XML {
+class Document;
+}
+
+namespace Mantid {
+namespace Kernel {
+class InternetHelper;
+}
+
+namespace DataHandling {
+namespace ISISJournal {
+/**
+ * ISISJournal: Helper class to aid in fetching ISIS specific run information
+ * from journal files
+ */
+
+class DLLExport ISISJournal {
+public:
+  using RunData = std::map<std::string, std::string>;
+
+  ISISJournal(std::string const &instrument, std::string const &cycle,
+              std::unique_ptr<Kernel::InternetHelper> internetHelper =
+                  std::make_unique<Kernel::InternetHelper>());
+  virtual ~ISISJournal();
```

Comment:
Can you mark destructor as default in the header, and remove it from the .cpp? In most projects if I don't see default or delete in the header I assume there is a custom implementation I need to take using in the .cpp

Ditto with the move constructor / assignment

Reply:
That causes a fun compilation error to do with forward declaring something in a unique_ptr...

Reply:
Oh right, it must be fixed in GCC-8 and 9 since I'm not seeing it in either builds. Either way I'm happy to leave it then